src/main/java/uk/gov/hmcts/ccd/domain/service/accessprofile/filter/matcher/AttributeMatcher.java

```diff
@@ -0,0 +1,19 @@
+package uk.gov.hmcts.ccd.domain.service.accessprofile.filter.matcher;
+
+import java.util.Optional;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.RoleAssignmentFilteringResult;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseDetails;
+
+public interface AttributeMatcher {
```

Comment:
implements AccessControl

Reply:
Looking at the implementation, I'm a bit concerned it's not generic enough to be used in the scenario when we are matching a case type rather that a case, where some of these matchers will have to just be ignored. 
Better to implement matching on a case type asap

Reply:
Will address the issues in https://tools.hmcts.net/jira/browse/RDM-11411, when we implement Case type filtering